arboretum/permissions/checks/test_org_permissions.py

```diff
@@ -0,0 +1,109 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Repository organization permissions check."""
+
+from compliance.check import ComplianceCheck
+from compliance.evidence import DAY, ReportEvidence, evidences
+from compliance.utils.data_parse import get_sha256_hash
+
+
+class OrgPermissionsCheck(ComplianceCheck):
+    """Monitor repository organization permissions."""
+
+    @property
+    def title(self):
+        """
+        Return the title of the checks.
+
+        :returns: the title of the checks
+        """
+        return 'Report on Repository Organization Permissions'
+
+    @classmethod
+    def setUpClass(cls):
+        """Initialize the check object with configuration settings."""
+        cls.config.add_evidences(
+            [
+                ReportEvidence(
+                    'org_permissions.md',
+                    'permissions',
+                    DAY,
+                    'Repository Permissions.'
+                )
+            ]
+        )
+        return cls
+
+    def test_org_permissions(self):
+        """Check the access to organization repositories."""
+        orgs = self.config.get('org.permissions.org_integrity.orgs')
+        for org in orgs:
+            host = org['url'].rsplit('/', 1)
+            service = 'gh'
+            if 'gitlab' in host:
+                service = 'gl'
+            elif 'bitbucket' in host:
+                service = 'bb'
+            url_hash = get_sha256_hash([org['url']], 10)
+            path = 'raw/permissions'
+            evidence_paths = {
+                'direct': f'{path}/{service}_direct_collaborators_'
+                + f'{url_hash}.json',
+                'outside': f'{path}/{service}_outside_collaborators_'
+                + f'{url_hash}.json',
+                'forks': f'{path}/{service}_forks_{url_hash}.json'
+            }
+            with evidences(self, evidence_paths) as ev:
+                self.add_failures(
+                    org['url'],
+                    self._check_collabs(
+                        ev['direct'].content_as_json,
+                        ev['outside'].content_as_json
+                    )
+                )
+                self.add_warnings(
+                    org['url'], self._check_forks(ev['forks'].content_as_json)
+                )
+
+    def _check_collabs(self, ev_direct, ev_outside):
+        repocollabs = []
+        for repo in ev_direct:
+            collabs = []
+            for c in ev_direct[repo]:
```

Comment:
This looks very odd to me.  It _seems_ like `ev_direct` is a dictionary where the key is `repo` and repo is a string?

Can you provide an example of what the evidence looks like here?  It seems like you can do ` for repo, c in ev_direct.items()` instead of this nested for loop.

Reply:
`ev_direct` is a dictionary and you are interested in both the key (the repo) and the value (the collaborator list) so you should be using the `items()` method in the for loop when looking through `ev_direct`.

L81 should be `for repo, collab_list in ev_direct.items():`

L83 would then be `for c in collab_list:`, but my preference is for it to be `for collab in collab_list:` since I'm not really sure why all of a sudden you switched to 1 letter variable names.

Reply:
Done